src/syrupy/serializers/base.py

```diff
@@ -93,10 +116,21 @@ def post_read(self, index: int = 0) -> None:
     def pre_write(self, data: "SerializableData", index: int = 0) -> None:
         self._ensure_snapshot_dir(index)
 
+    @final
     def write(self, data: "SerializableData", index: int = 0) -> None:
+        """
+        Override _write_snapshot_or_remove_file in subclass to change behaviour
+        """
         snapshot_file = self.get_filepath(index)
         snapshot_name = self.get_snapshot_name(index)
-        self.write_snapshot_or_remove_file(snapshot_file, snapshot_name, data)
+        if self.test_location.testname not in snapshot_name:
+            print(
+                warning_style(
+                    f"Warning: snapshot name '{snapshot_name}' does not"
+                    f" contain testname '{self.test_location.testname}'"
```

Comment:
Added because this relies on the test names for deciding which tests were run by pytest collection